### ISSUE TITLE
Compatibility with the Discord Slack-Compatible Webhook API

### ISSUE THREAD

**Issue by apkd** [Narochno/Narochno.Serilog.Slack#8]
First of all, thanks for publishing this package - it's gonna make monitoring a lot more fun.

That said, I've been playing around and I've noticed the sink doesn't work with the [Discord Slack-Compatible Webhook API](https://discordapp.com/developers/docs/resources/webhook#execute-slackcompatible-webhook). I imagine this probably isn't a priority, but I figured I'd let you know either way.

According to the docs, they

> do not support Slack's channel, icon_emoji, mrkdwn, or mrkdwn_in properties.

which may have something to do with the error. However, I tested the API with a [different Slack sink package](https://github.com/marcio-azevedo/serilog-sinks-slack) and that one worked without issue.

Sample failing program and exception log:
https://gist.github.com/apkd/58b3eba36ef9511c8fd7c9cc4399e49b

**Comment by alanedwardes** [comment]
Hi apkd,

It looks like you've hit two things - one is the bad error handling, and another is the Discord compatibility. Both of these issues I've raised on the underlying Slack client here: https://github.com/Narochno/Narochno.Slack/issues/4 & https://github.com/Narochno/Narochno.Slack/issues/5

If you subscribe to those issues, they'll get closed when the Slack client is fixed. We'll close this issue when we have updated the Slack client in this package + verified.

Thanks!

**Comment by alanedwardes** [comment]
@apkd this has been addressed in version `1.3.0` of Narochno.Serilog.Slack, https://github.com/Narochno/Narochno.Serilog.Slack/pull/9.

To set it up, use the following Sink extension:
```csharp
Log.Logger = new LoggerConfiguration()
   .WriteTo.Slack(new SlackConfig { WebHookUrl = "<your discord webhook>/slack" }, new AttachmentsSlackFormatter())
   .CreateLogger();
```

You can also make your own Slack message formatter by implementing `ISlackFormatter` and passing it there.

I will close this for now, feel free to re-open or open a new issue if you find that it isn't working.

Alan
